### PR TITLE
Remove 'oldPowershellExeName' contains powershell.exe

### LINKED ISSUE #5135: Remove 'oldPowershellExeName' from 'RemoveSxSPsHomeModulePath' before RC/GA
Since we rename `powershell` to `pwsh`, the variables `oldPowershellExeName` and the use of it should be removed from [`RemoveSxSPsHomeModulePath`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs#L768) before RC/GA.

It's kept for now because for now people will try opening `pwsh` from existing powershell core.

### PR DESCRIPTION
**Pull request by chunqingchen**
This is to resolve #5135

### CODE REVIEW

**Comment by chunqingchen** [comment]
@daxian-dbw hi dongbo, could you please take a look?

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs [hunk 1]
```diff
@@ -769,10 +769,8 @@ private static string RemoveSxSPsHomeModulePath(string currentProcessModulePath,
         {
 #if UNIX
             const string powershellExeName = "pwsh";
-            const string oldPowershellExeName = "powershell";
 #else
             const string powershellExeName = "pwsh.exe";
-            const string oldPowershellExeName = "powershell.exe";
 #endif
             const string powershellDepsName = "pwsh.deps.json";
             const string oldPowershellDepsName = "powershell.deps.json";
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: old deps file should be removed.

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs [hunk 2]
```diff
@@ -796,10 +794,9 @@ private static string RemoveSxSPsHomeModulePath(string currentProcessModulePath,
                 {
                     string parentDir = Path.GetDirectoryName(trimedPath);
                     string psExePath = Path.Combine(parentDir, powershellExeName);
-                    string oldExePath = Path.Combine(parentDir, oldPowershellExeName);
                     string psDepsPath = Path.Combine(parentDir, powershellDepsName);
                     string oldDepsPath = Path.Combine(parentDir, oldPowershellDepsName);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: old deps file.

**Review by daxian-dbw — DISMISSED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@adityapatwardhan Can you please take a look at this PR? Thanks!

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**